typing/ctype.ml

```diff
@@ -3269,6 +3269,15 @@ and mcomp_row type_pairs env row1 row2 =
       | _ -> ())
     pairs
 
+and mcomp_unsafe_mode_crossing umc1 umc2 =
```

Comment:
`mcomp` is symmetric. I think it should do the same check as the one in includecore. (I'm actually not sure why the checks on type_kinds are implemented separately between includecore and here.)